lib/vrl/parser/src/parser.lalrpop

```diff
@@ -255,13 +255,15 @@ AssignmentOp: AssignmentOp = {
 AssignmentSingle: Assignment = {
     <target: Sp<AssignmentTarget>>
         <op: AssignmentOp>
+        NonterminalNewline*
         <expr: Box<Expr>> => Assignment::Single { target, op, expr },
 }
 
 AssignmentInfallible: Assignment = {
     <ok: Sp<AssignmentTarget>> ","
         <err: Sp<AssignmentTarget>>
         <op: AssignmentOp>
+        NonterminalNewline*
```

Comment:
Just a tangential question, but why is `NonterminalNewline` needed instead of using the `Newline` terminal directly?

Reply:
Good question 😄 I _believe_ at some point `NonterminalNewline` was more than just `"\n"`, but I also added it to make it more clear what this newline represents (e.g. it doesn't start a new expression).

We could remove it in the future, I'm somewhat ambivalent on that.